src/WaveTrack.cpp

```diff
@@ -1694,12 +1695,59 @@ void WaveTrack::Disjoin(double t0, double t1)
          }
       }
    }
+   /* This should use proper trim instead of soft SplitDelete
+    * Creates new clips from the segments copying from the original clip,
+    * than remove the original clip when it is no longer has impacted regions
+    */
+   
+   sampleCount lastEndPos = -1;
+   WaveClip *newClip;
+   for (unsigned int i = 0; i < clipRegions.size(); i++)
+   {
+      const ClipRegion& region = clipRegions.at(i);
+      // There is no segment to be kept on the left
+      if (region.clip->GetPlayStartSample() == region.start)
+      {
+         // the region spans to the end of the clip, so we can just delete it
+         if (region.clip->GetPlayEndSample() == region.end) {
+            WaveClipHolder toRemove = RemoveAndReturnClip(region.clip.get());
+            toRemove.~shared_ptr();
```

Comment:
No need to explicitly call the destructor.  The destructor will be called when toRemove goes out of scope.

Reply:
Removed explicit destructor call